external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/PostgresKrbIntegrationSuite.scala

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.jdbc
+
+import com.spotify.docker.client.messages.{ContainerConfig, HostConfig}
+import java.io.{File, FileInputStream, FileOutputStream}
+import java.sql.Connection
+import java.util.Properties
+import javax.security.auth.login.Configuration
+import org.apache.hadoop.minikdc.MiniKdc
+import scala.io.Source
+
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.execution.datasources.jdbc.connection.PostgresConnectionProvider
+import org.apache.spark.sql.types.StringType
+import org.apache.spark.tags.DockerTest
+import org.apache.spark.util.{SecurityUtils, Utils}
+
+@DockerTest
+class PostgresKrbIntegrationSuite extends DockerJDBCIntegrationSuite {
```

Comment:
It seems that we are going to add `OracleKrbIntegrationSuite` and `MsSQLServerKrbIntegrationSuite`. Could you make a base class for them from the beginning?

Reply:
Sure, we can make the initial steps but since we don't know all the details now it will be probably modified. Doing it...